crates/matrix-sdk-base/src/media.rs

```diff
@@ -86,211 +76,93 @@ impl UniqueKey for MediaType {
-                .or_else(|| info.thumbnail_url.as_ref().map(|uri| MediaType::Uri(uri.clone())))
-        })
+    fn thumbnail_src(&self) -> Option<MediaSource> {
+        self.info.as_ref()?.thumbnail_src.clone()
     }
 }
 
 impl MediaEventContent for ImageMessageEventContent {
-    fn file(&self) -> Option<MediaType> {
-        self.file
-            .as_ref()
-            .map(|e| MediaType::Encrypted(e.clone()))
-            .or_else(|| self.url.as_ref().map(|uri| MediaType::Uri(uri.clone())))
+    fn src(&self) -> Option<MediaSource> {
+        Some(self.src.clone())
     }
 
-    fn thumbnail(&self) -> Option<MediaType> {
+    fn thumbnail_src(&self) -> Option<MediaSource> {
         self.info
             .as_ref()
-            .and_then(|info| {
-                info.thumbnail_file
-                    .as_ref()
-                    .map(|file| MediaType::Encrypted(file.clone()))
-                    .or_else(|| info.thumbnail_url.as_ref().map(|uri| MediaType::Uri(uri.clone())))
-            })
-            .or_else(|| self.url.as_ref().map(|uri| MediaType::Uri(uri.clone())))
+            .and_then(|info| info.thumbnail_src.clone())
+            .or_else(|| Some(self.src.clone()))
     }
 }
 
 impl MediaEventContent for VideoMessageEventContent {
-    fn file(&self) -> Option<MediaType> {
-        self.file
-            .as_ref()
-            .map(|e| MediaType::Encrypted(e.clone()))
-            .or_else(|| self.url.as_ref().map(|uri| MediaType::Uri(uri.clone())))
+    fn src(&self) -> Option<MediaSource> {
+        Some(self.src.clone())
     }
 
-    fn thumbnail(&self) -> Option<MediaType> {
+    fn thumbnail_src(&self) -> Option<MediaSource> {
         self.info
             .as_ref()
-            .and_then(|info| {
-                info.thumbnail_file
-                    .as_ref()
-                    .map(|file| MediaType::Encrypted(file.clone()))
-                    .or_else(|| info.thumbnail_url.as_ref().map(|uri| MediaType::Uri(uri.clone())))
-            })
-            .or_else(|| self.url.as_ref().map(|uri| MediaType::Uri(uri.clone())))
+            .and_then(|info| info.thumbnail_src.clone())
+            .or_else(|| Some(self.src.clone()))
     }
 }
 
 impl MediaEventContent for LocationMessageEventContent {
-    fn file(&self) -> Option<MediaType> {
+    fn src(&self) -> Option<MediaSource> {
         None
     }
 
-    fn thumbnail(&self) -> Option<MediaType> {
-        self.info.as_ref().and_then(|info| {
-            info.thumbnail_file
-                .as_ref()
-                .map(|file| MediaType::Encrypted(file.clone()))
-                .or_else(|| info.thumbnail_url.as_ref().map(|uri| MediaType::Uri(uri.clone())))
-        })
-    }
-}
-
-#[cfg(all(feature = "encryption", test))]
-pub(crate) mod test {
-    use ruma::events::room::{
-        message::{FileInfo, LocationInfo, VideoInfo},
-        ImageInfo,
-    };
-
-    use super::*;
-
-    fn encrypted_test_data() -> (Box<MxcUri>, EncryptedFile) {
```

Comment:
I assume some variant of those tests are now part of Ruma?

Reply:
I'll make sure we have them.